lib/Common/Memory/LargeHeapBlock.cpp

```diff
@@ -280,60 +286,112 @@ LargeHeapBlock::ReleasePagesShutdown(Recycler * recycler)
 #endif
 }
 
-template void LargeHeapBlock::ReleasePagesSweep<true>(Recycler * recycler);
-template void LargeHeapBlock::ReleasePagesSweep<false>(Recycler * recycler);
-
-template<bool pageheap>
 void
 LargeHeapBlock::ReleasePagesSweep(Recycler * recycler)
 {
     recycler->heapBlockMap.ClearHeapBlock(this->address, this->pageCount);
 
-    ReleasePages<pageheap>(recycler);
+    ReleasePages(recycler);
 }
 
+#ifdef RECYCLER_PAGE_HEAP
+__declspec(noinline)
+void LargeHeapBlock::VerifyPageHeapPattern()
+{
+    Assert(InPageHeapMode());
+    if (this->allocAddressEnd > this->address) // in case OOM while allocate heapBlockMap
+    {
+#if _M_X64
+        byte* addr = (byte*)this->allocAddressEnd;
+        uint64* alignAddr = (uint64*)Math::Align(((uint64)addr), sizeof(void*));
+
+        for (int i = 0; addr+i<(byte*)alignAddr; i++)
+        {
+            byte current = addr[i];
+            if (current != 0xF0u)
+            {
+                Assert(false);
+                ReportFatalException(NULL, E_FAIL, Fatal_Recycler_MemoryCorruption, 2);
+            }
+        }
+
+        for (int i = 0; alignAddr + i < (uint64*)this->allocAddressEnd; i++)
```

Comment:
What is this second loop for? Isn't alignedAddr always >= allocAddressEnd (since it's basically allocAddressEnd aligned to the next pointer boundary?)